enhancements/insights/config-api.md

```diff
@@ -0,0 +1,296 @@
+
+### Risks and Mitigations
+
+Risk: Insights operator is not able to read the new config API.
+
+Mitigation: The Insights operator is marked as unvailable through the corresponding clusteroperator condition.
+
+Risk: Somebody/something is repeatedly updating the `ForceGatheringReason` attribute and thus forcing gathering. There is a risk that 
+the cluster would upload data too often, but this assumes that the "repeating interval" allows uploading (if it's forced too quickly then there will be no upload, because the gathering is not over).
+
+Mitigation: Maybe we should introduce some limit on the number of forced gatherings within 1 hour, for example. 
+
+## Design Details
+
+### On demand data gathering options
+
+The on demand data gathering can happen almost immediately when there is no active gathering in the progress, but when there is some in progress 
+then the question is what to do in this situation. There are following options:
+
+- schedule it - e.g using something like the `ScheduleGatheringAt` attribute where the user would provide some time in the future.
+  - Pros 
+    - no need to interrupt gathering in progress
+    - we could probably use the `RateLimiting` queue and add a new gathering to the queue with the `AddAfter` method. Otherwise we would need to define some waiting time (when there’s already some gathering in progress) and right now it’s difficult to estimate an average gathering time since the gatherers run in parallel
+  - Cons
+    - user has to define the scheduling time correctly in the future
+    - gathering doesn’t happen immediately
+- force it - e.g using a `ForceGatherinReason` attribute where the user would provide some string as reason (mentioned in the [Proposal](#proposal)). 
+  - pros
+    - [roviding a reason (e.g “Refresh Insights”) is simpler for configuration than providing a time
+    - gathering would be triggered almost immediately
+    - there is already similar precedent with `ForceRedeploymentReason` when redeploying static pods
+  - cons
+    - need to be able to interrupt gathering (if any in progress)
+
+### Open Questions
+
+- ~~Can the API define default values? How does it work or how is it implemented?~~ This is done by using `// +kubebuilder:default=`
+- Should the new time of forced gathering (if any defined by user) set a new time for the existing gathering interval? If I force gathering at time A, will the new automatic gathering be in time A + interval?
+
+### Test Plan
+
+The new config API will require updates of the existing Insights operator testsuite. The testsuite will be extended to cover the new API and respective attributes. There will be a new test checking the forced gathering feature. 
+
+### Graduation Criteria
+
+The plan is to introduce the first version of the new API as the `v1alpha1` version and once we gain some confidence and experience with the new API then
+we can move it to `v1beta1`.
```

Comment:
Introduce v1alpha1 as GA enabled by default or is this only enabled via a feature gate until it graduates? If it's enabled by default then even if you call it v1alpha1 I believe you're still obligated to migrate to later api versions seamlessly.

Reply:
Introducing a giant API as v1alpha1 does not sound like a winning strategy. Start small, super small. Add on-demand.

Note that we are super restricted of how to evolve APIs of the same GVR. Versioning does not help much, as @sdodson writes: as soon as non-feature-gate users have this API persisted we are mostly stuck with all the field names, types, formats, validation, etc.

Reply:
OK thanks. This is something I wasn't sure about at all so if I understand it correctly you're suggesting to introduce it as `v1alpha1` but with feature gate right? I think I am fine with that...would it be `TechPreviewNoUpgrade` (my feeling is that almost nobody enables it) or could it be another feature?

Reply:
I narrowed down the API proposal and added the info about `TechPreviewNoUpgrade` feature gate. Can I resolve this discussion? WDYT please?

Reply:
The API was reduced heavily and will be behind TP feature gate. Resolving this discussion.